app/org/maproulette/controllers/api/DataController.scala

```diff
@@ -183,19 +185,67 @@ class DataController @Inject()(sessionManager: SessionManager, challengeDAL: Cha
         ))
       } else {
         SearchParameters.withSearch { implicit params =>
-          Ok(Json.toJson(
-            this.dataManager.getChallengeSummary(challengeId = Some(id), priority = Utils.toIntList(priority), params = Some(params))
-          ))
+          val response = this.dataManager.getChallengeSummary(challengeId = Some(id), priority = Utils.toIntList(priority), params = Some(params))
+
+          if (includeByPriority) {
```

Comment:
This block of code is duplicated, so you should move it into a function with some input parameters to handle the input parameters. The Utils.insertIntoJson section would obviously not be included in that.

Reply:
You are right. Extracted it into a separate method.